### PR TITLE
WLTS-33 - Update Swagger config

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/WLTS-33

### Change description ###

Enrich swagger UI pages, fix try it now functionality so that Authorization and ServiceAuthorization headers can be set globally, and publish swagger output to https://hmcts.github.io/reform-api-docs/

> Note: PR hmcts/reform-api-docs#270 is dependent on this PR being merged into master and the new Publish Swagger Specs workflow task to have completed OK.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java [hunk 1]
```diff
@@ -35,39 +40,53 @@ public DictionaryController(DictionaryService dictionaryService) {
...
-        @ApiResponse(responseCode = "500", description = "Error occurred on the server")
-    })
-    public void putDictionary(@RequestBody Dictionary dictionaryRequest,
-                              @RequestHeader(SERVICE_AUTHORIZATION) String clientS2SToken) {
+    @Operation(summary = "Upload a set of phrases for which translations may be provided.",
+        description = "If user's current IDAM role is `manage-translations`\n\n"
+            + "\t - User can submit English phrases with/without corresponding Welsh translations\n\n"
+            + "If user's current IDAM role is `load translations`\n\n"
+            + "\t - User can submit English phrases only\n\n"
+            + "If calling service is on approved list of services for this endpoint\n\n"
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java:62]: May need expanding: e.g.

```
+ "If calling service is on approved list of services for this endpoint that don't need to supply Authorization header: then\n\n"
```

**Review by mattnayler — APPROVED** [review]
Approved:  however I have included some suggested text changes to include extra error messages highligted in WLTS-27 #51.

File: src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java [hunk 2]
```diff
@@ -35,39 +40,53 @@ public DictionaryController(DictionaryService dictionaryService) {
...
-        @ApiResponse(responseCode = "401", description = "Unauthorised"),
-        @ApiResponse(responseCode = "403", description = "Forbidden")
-    })
+    @Operation(summary = "Download a set of phrases potentially associated with a case type for which translations "
+        + "will be eventually added.",
+        description = "Users calling this endpoint must have the `" + MANAGE_TRANSLATIONS_ROLE + "` role",
+        responses = {
+            @ApiResponse(responseCode = "200", description = "Dictionary returned successfully"),
+            @ApiResponse(responseCode = "401", description = AUTHENTICATION_TOKEN_INVALID, content = @Content()),
+            @ApiResponse(responseCode = "403", description = UNAUTHORISED_S2S_SERVICE, content = @Content())
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java]: Fixed in https://github.com/hmcts/ts-translation-service/pull/54/commits/95363debf60125f51fd01c8290766ad05c426af5

I can't use the constant as it is a formatted string, and you can't do `String.format(MY_CONSTANT, "value")` here - it has to be a constant

File: src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java [hunk 3]
```diff
@@ -35,39 +40,53 @@ public DictionaryController(DictionaryService dictionaryService) {
...
     @PostMapping(path = TRANSLATIONS_URL, consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
-    @ApiResponses(value = {
-        @ApiResponse(responseCode = "200", description = "Translation returned successfully"),
-        @ApiResponse(responseCode = "400", description = TranslationsRequest.BAD_REQUEST_MESSAGE),
-        @ApiResponse(responseCode = "401", description = "Unauthorised"),
-        @ApiResponse(responseCode = "403", description = "Forbidden")
-    })
+    @Operation(summary = "Get the Welsh translation of the provided set of English phrases.",
+        description = "User does not require any specific roles to call this endpoint",
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java]: Fixed in https://github.com/hmcts/ts-translation-service/commit/95363debf60125f51fd01c8290766ad05c426af5

File: src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java [hunk 4]
```diff
@@ -35,39 +40,53 @@ public DictionaryController(DictionaryService dictionaryService) {
...
+        responses = {
+            @ApiResponse(responseCode = "200", description = "Translation returned successfully"),
+            @ApiResponse(responseCode = "400", description = TranslationsRequest.BAD_REQUEST_MESSAGE),
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/translate/controllers/DictionaryController.java]: Fixed in https://github.com/hmcts/ts-translation-service/commit/95363debf60125f51fd01c8290766ad05c426af5

**Review by mattnayler — APPROVED** [review]
✅ re-approved after chaining onto WLTS-27 branch / PR: #51.

**Pull request merged**